src/Storages/VirtualColumnUtils.cpp

```diff
@@ -189,6 +213,21 @@ ColumnPtr getFilterByPathAndFileIndexes(const std::vector<String> & paths, const
     return block.getByName("_idx").column;
 }
 
+std::map<std::string, std::string> parsePartitionMapFromPath(const std::string & path)
```

Comment:
This code is almost the same as in `parseVirtualColumnNameFromPath`, consider implementing single function that can be used in both places